src/pudl/analysis/ferc1_eia.py

```diff
@@ -431,596 +436,139 @@ def get_features(self, clobber=False):
+        {
+            "solver": ["saga"],
+            "C": [1000, 1, 10, 100],
+            "class_weight": [None, "balanced"],
+            "penalty": ["elasticnet"],
+            "l1_ratio": [0.1, 0.3, 0.5, 0.7, 0.9],
+        },
+    ]
+    x_train = features_train.to_numpy()
+    y_train = np.where(
+        features_train.merge(
+            train_df,
+            how="left",
             left_index=True,
             right_index=True,
-            how="left",
-        )
-        return predict_all_df
+            indicator=True,
+        )["_merge"]
+        == "both",
+        1,
+        0,
+    )
+    lrc = LogisticRegression()
+    clf = GridSearchCV(estimator=lrc, param_grid=param_grid, verbose=True, n_jobs=-1)
+    clf.fit(X=x_train, y=y_train)
+    y_pred = clf.predict(x_train)
+    precision, recall, f_score, _ = precision_recall_fscore_support(
+        y_train, y_pred, average="binary"
+    )
+    accuracy = clf.best_score_
+    logger.info(
+        "Scores from the best model:\n"
+        f"    Accuracy:  {accuracy:.02}\n"
+        f"    F-Score:   {f_score:.02}\n"
+        f"    Precision: {precision:.02}\n"
+        f"    Recall:    {recall:.02}\n"
+    )
+    preds = clf.predict(features_all.to_numpy())
+    probs = clf.predict_proba(features_all.to_numpy())
+    final_df = pd.DataFrame(
+        index=features_all.index, data={"match": preds, "prob_of_match": probs[:, 1]}
+    )
+    match_df = final_df[final_df.match == 1].sort_values(
+        by="prob_of_match", ascending=False
+    )
+    return match_df
 
-    def get_coefs(self):
-        """Get the best models coeficients.
 
-        The coeficients are the measure of relative importance that the model determined
-        that each feature vector.
-        """
-        self.coefs = self.best.loc[0, "coef"]
-        return self.coefs
-
-    def get_best_matches(self, features_train, features_all):
-        """Run logistic regression model and choose highest scoring matches.
-
-        From :meth:`TuneModel.test_model_parameters()`, we get a dataframe in which each
-        record correspondings to one model run and contains info like scores of the run
-        (how well it predicted our training data). This method takes the result from
-        :meth:`TuneModel.test_model_parameters()` and choose the model hyperparameters
-        that scored the highest.
-
-        The logistic regression model this method employs returns all matches from the
-        candiate matches in ``features_all``. But we only want one match for each FERC1
-        plant record. So this method uses the coeficients from the model (which are a
-        measure of the importance of each of the features/columns in ``features_all``)
-        so weight the feature vecotrs. With the sum of the weighted feature vectors for
-        each model match, this method selects the hightest scoring match via
-        :meth:`calc_best_matches()`.
+def remove_murky_matches(match_df, difference_threshold):
```

Comment:
Previously, you were choosing a best match based on weighting the feature vectors by their coefficients and finding the interquartile range. Do you remember if there was a reason to do this method for selecting the best FERC to EIA match?

Here, I thought I'd experiment a little with a simpler method. For each FERC record, I choose the highest probability EIA record match as long as the difference between that record and the next highest probability is over a set threshold. But happy to go back to the interquartile range. It might perform slightly better.

Reply:
I think the answer to this also depends on how certain you want matches to be. And that can also be dictated by the thresholds that are set.

Reply:
This function could also be renamed `calc_best_matches` to be consistent with what it used to be called. In the previous version of `calc_best_matches` there were a bunch of stats that were printed out that I was confused about (the Winning match stats section with matches vs ferc, best match vs ferc etc.) I'm not currently printing those stats but I can put them back in.